utils/add_platform_rule.py

```diff
@@ -0,0 +1,321 @@
+    if shutil.which('oc') is None:
+        print('oc is required to test a rule in-cluster.')
+        return 1
+
+    print('testing rule %s in-cluster' % args.rule)
+
+    build_cmd = 'utils/build_ds_container.sh ocp4'
+    profile_path = 'ocp4/profiles/test.profile'
+
+    if not os.path.exists(PLATFORM_RULE_DIR + '/' + args.rule):
+        print('no rule for %s, run "create" first' % args.rule)
+        return 1
+
+    if args.deploy:
+        print('deploying compliance-operator')
+        subprocess.getstatusoutput("utils/deploy_compliance_operator.sh")
+
+    # create a solo profile for the rule
+    with open(profile_path, 'w') as file:
+        file.write(PROFILE_TEMPLATE.format(RULE_NAME=args.rule))
+
+    # execute the build_ds_container script
+    print('pushing image build to cluster')
+    ret_code, output = subprocess.getstatusoutput(build_cmd)
+    if ret_code != 0:
+        print(output)
+        try:
+            os.remove(profile_path)
+        except OSError:
+            pass
+        return 1
+
+    ret_code, output = subprocess.getstatusoutput(
+        'oc delete compliancescans/test')
+    if ret_code == 0:
+        # if previous compliancescans were actually deleted, wait a bit to allow resources to clean up.
+        print('waiting for cleanup from a previous test run')
+        time.sleep(20)
+
+    # create a single-rule scan
+    profile = 'xccdf_org.ssgproject.content_profile_test'
+    apply_cmd = ['oc', 'apply', '-f', '-']
+    with subprocess.Popen(apply_cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE) as proc:
+        out, err = proc.communicate(
+            input=TEST_SCAN_TEMPLATE.format(PROFILE=profile).encode())
+        if proc.returncode != 0:
+            print('error applying scan object: %s' % err)
+            try:
+                os.remove(profile_path)
+            except OSError:
+                pass
+            return 1
+
+    # poll for the DONE result
+    timeout = time.time() + 60   # A minute is generous for the platform scan.
+    scan_result = None
+    while True:
+        ret_code, output = subprocess.getstatusoutput(
+            'oc get compliancescans/test -o template="{{.status.phase}} {{.status.result}}"')
+        if output is not None:
+            print('output from last phase check: %s' % output)
+        if output.startswith('DONE'):
+            scan_result = output[5:]
+            break
+        if time.time() >= timeout:
+            break
+        time.sleep(1)
+
+    if scan_result == None:
+        print('timeout waiting for scan to finish')
+        return 1
+
+    print(scan_result)
+    return 0
+
+
+def testFunc(args):
+    if shutil.which('podman') is None:
+        print('podman is required')
+        return 1
+
+    print('testing rule %s locally' % args.rule)
+    ret_code, out = subprocess.getstatusoutput(
+        'podman run -it -v %s:/content/:z -v %s:/tmp/:z %s %s' % (args.file, args.objectdir, OSCAP_TEST_IMAGE, OSCAP_CMD_OPTS))
```

Comment:
I'm having issues with this command. It seems like the objectdir is actually required, the default `/tmp` doesn't work for me, I'm getting `Error: relabel failed "/tmp": SELinux relabeling of /tmp is not allowed`. Also, just using the file as the volume doesn't work either, I'm getting errors like `Error: error creating named volume "build/ssg-ocp4-ds.xml": error running volume create option: names must match [a-zA-Z0-9][a-zA-Z0-9_.-]*: invalid argument`. Did you maybe want to share the parent directory of the DS file instead?

Reply:
Also the test profile doesn't exist by default..